src/pages/announcements/CreateAnnouncementDialog.tsx

```diff
@@ -0,0 +1,87 @@
+/*
+ * Copyright (c) 2024, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+import { FunctionComponent, useCallback } from 'react';
+import {
+    Dialog,
+    DialogActions,
+    DialogContent,
+    DialogTitle,
+    Grid,
+} from '@mui/material';
+import { FormattedMessage } from 'react-intl';
+import { FormProvider, useForm } from 'react-hook-form';
+import { yupResolver } from '@hookform/resolvers/yup';
+import { DurationInput } from './DurationInput';
+import { SubmitButton, TextInput } from '@gridsuite/commons-ui';
+import {
+    AnnouncementFormData,
+    emptyFormData,
+    formSchema,
+    fromFrontToBack,
+    MESSAGE,
+} from './utils';
+import { UserAdminSrv } from '../../services';
+
+interface CreateAnnouncementDialogProps {
+    open: boolean;
+    onClose: () => void;
+}
+
+export const CreateAnnouncementDialog: FunctionComponent<
+    CreateAnnouncementDialogProps
+> = (props) => {
+    const formMethods = useForm<AnnouncementFormData>({
+        defaultValues: emptyFormData,
+        //@ts-ignore because yup TS is broken
+        resolver: yupResolver(formSchema),
+    });
+
+    const { handleSubmit, reset } = formMethods;
+
+    const onSubmit = useCallback(
+        (formData: AnnouncementFormData) => {
+            UserAdminSrv.createAnnouncement(fromFrontToBack(formData));
+            reset();
+            props.onClose();
+        },
+        [props, reset]
+    );
+
+    return (
+        //@ts-ignore because RHF TS is broken
+        <FormProvider validationSchema={formSchema} {...formMethods}>
```

Comment:
Are you sure ? It's not just that we shouldn't define this property ? Schema is in the yup resolver already, no ?

Reply:
It's something we already talked about on Slack.
We found some way to avoid the issue, ie
https://github.com/gridsuite/gridstudy-app/blob/760264391c6b49a26e60edb94973b5752dbf9d67/src/components/dialogs/dynamicsimulation/event/dynamic-simulation-event-dialog.tsx#L228-L233
But not working here

Reply:
Actually I'm wondering if we don't do it wrong since the start haha